### PR TITLE
Soccer Twos Environment

### PR DESCRIPTION
**Pull request by awjuliani**
* Add `Soccer Twos` environment, along with training parameters, embedded model, and relevant documentation.

### CODE REVIEW

File: docs/Learning-Environment-Examples.md [hunk 1]
```diff
@@ -198,3 +198,29 @@ If you would like to contribute environments, please see our
     * Vector Action space: (Continuous) 3 corresponding to agent force applied for the jump.
     * Visual Observations: None
 * Reset Parameters: None
+
+## Soccer Twos
+
+![SoccerTwos](images/soccer.png)
+
+* Set-up: Environment where four agents compete in a 2 by 2 toy soccer game.
```
Comment (mmattar) [inline docs/Learning-Environment-Examples.md]: 2-vs-2

File: docs/Learning-Environment-Examples.md [hunk 2]
```diff
@@ -198,3 +198,29 @@ If you would like to contribute environments, please see our
...
+    * Striker:
+        * +1 For scoring in opponent's goal.
+        * -0.1 For goal entering own team's goal.
+        * -0.001 Existential penalty.
+    * Goalie:
+        * -1 When scored on.
+        * +0.1 When scores goal.
+        * +0.001 Existential bonus.
+* Brains: Two brain with the following observation/action space:
+    * Vector Observation space: (Continuous) 112 corresponding to local ray-cast perception in 180 degree view from front of agent.
```
Comment (mmattar) [inline docs/Learning-Environment-Examples.md]: Explain where the 112 come from

File: docs/Learning-Environment-Examples.md [hunk 3]
```diff
@@ -198,3 +198,29 @@ If you would like to contribute environments, please see our
...
+    * Goalie: Prevent the ball from entering its own goal.
+* Agents: The environment contains four agents, with two linked to one brain (strikers) and two linked to another (goalies).
+* Agent Reward Function (dependent):
+    * Striker:
+        * +1 For scoring in opponent's goal.
+        * -0.1 For goal entering own team's goal.
+        * -0.001 Existential penalty.
+    * Goalie:
+        * -1 When scored on.
+        * +0.1 When scores goal.
```
Comment (mmattar) [inline docs/Learning-Environment-Examples.md]: The team or the goalie itself

File: unity-environment/ProjectSettings/ProjectSettings.asset [hunk 4]
```diff
@@ -555,13 +535,15 @@ PlayerSettings:
   webGLTemplate: APPLICATION:Default
   webGLAnalyzeBuildSize: 0
   webGLUseEmbeddedResources: 0
-  webGLUseWasm: 0
   webGLCompressionFormat: 1
+  webGLLinkerTarget: 0
   scriptingDefineSymbols:
-    1: 
+    1: ENABLE_TENSORFLOW
```
Comment (mmattar) [inline unity-environment/ProjectSettings/ProjectSettings.asset]: should we disable by default?

File: unity-environment/Assets/ML-Agents/Examples/Soccer/Scripts/SoccerBallController.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections;
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/Soccer/Scripts/SoccerBallController.cs]: for this file ensure that tabs are saved as spaces.

**Comment by mmattar** [comment]
Should we rename environment folder from Soccer to SoccerTwos?

**Comment by awjuliani** [comment]
@mmattar My thinking is we will have separate scenes which correspond to different combinations of agents, with the overall folder being "soccer."

**Comment by nanxintin** [comment]
Hi, I want to figure out what the state vector in soccer2 environment means. 
I found some related codes：
if (team == Team.red)：
     detectableObjects = new string[] { "ball", "redGoal", "blueGoal", "wall", "redAgent", "blueAgent" };
could you tell me what these objects mean?
if I control a red striker, what are its detectable objects?

**Pull request merged**